web/src/types.ts

```diff
@@ -97,6 +97,7 @@ export interface ZoneDetail extends ZoneOverview {
   completenessPercentage?: number;
   // Capacity is only available on hourly details
   capacity?: { [key in ElectricityModeType]: number | null };
+  capacitySources?: { [key in ElectricityModeType]: string };
```

Comment:
This could probably also potentially be null in some cases, better include that than exclude it.